assets/js/googlesitekit/data/create-settings-store.js

```diff
@@ -280,23 +283,25 @@ export const createSettingsStore = (
 		 *
 		 * @since 1.6.0
 		 * @since 1.77.0 Added ability to filter settings using `keys` argument.
+		 * @since n.e.x.t Added makeDefaultHaveSettingsChanged for default selector and added haveSettingsChanged custom callback as an optional parameter.
 		 *
 		 * @param {Object}     state Data store's state.
 		 * @param {Array|null} keys  Settings keys to check; if not provided, all settings are checked.
 		 * @return {boolean} True if the settings have changed, false otherwise.
 		 */
-		haveSettingsChanged( state, keys = null ) {
-			const { settings, savedSettings } = state;
+		haveSettingsChanged: createRegistrySelector(
+			( select ) => ( state, keys ) => {
+				if ( haveSettingsChanged ) {
+					return haveSettingsChanged( { state, keys, select } );
+				}
 
-			if ( keys ) {
-				return ! isEqual(
-					pick( settings, keys ),
-					pick( savedSettings, keys )
-				);
+				return makeDefaultHaveSettingsChanged( {
+					state,
+					keys,
+					select,
+				} );
 			}
-
-			return ! isEqual( settings, savedSettings );
-		},
+		),
```

Comment:
Let's use the same approach as the `createValidationSelector` function uses. It should be something like this:

```js
haveSettingsChanged: createRegistrySelector( ( select ) => ( state, ...args ) => validateHaveSettingsChanged( select, state, ...args ) );
```